src/test/java/htsjdk/samtools/AbstractBAMFileIndexTest.java

```diff
@@ -60,4 +64,46 @@ public int read() throws IOException {
         buffer.readInteger();
         buffer.readBytes(new byte[10000]);
     }
+
+    @Test
+    public static void testGetNumIndexLevels() {
+        Assert.assertEquals(AbstractBAMFileIndex.getNumIndexLevels(), 6);
+    }
+
+    @Test
+    public static void testGetFirstBinInLevelOK() {
+        Assert.assertEquals(AbstractBAMFileIndex.getFirstBinInLevel(0), 0);
+        Assert.assertEquals(AbstractBAMFileIndex.getFirstBinInLevel(1), 1);
+        Assert.assertEquals(AbstractBAMFileIndex.getFirstBinInLevel(2), 9);
+        Assert.assertEquals(AbstractBAMFileIndex.getFirstBinInLevel(3), 73);
+        Assert.assertEquals(AbstractBAMFileIndex.getFirstBinInLevel(4), 585);
+        Assert.assertEquals(AbstractBAMFileIndex.getFirstBinInLevel(5), 4681);
+    }
+
+    @Test (expectedExceptions = SAMException.class)
+    public static void testGetFirstBinInLevelFail() {
+        AbstractBAMFileIndex.getFirstBinInLevel(6);
+    }
+
+    @Test
+    public static void testGetLevelSizeOK() {
+
+        final AbstractBAMFileIndex afi = new DiskBasedBAMFileIndex(new File(BAM_FILE.getPath() + ".bai"),
```

Comment:
Unnecessary `getPath()` here and below, you can just write `new File(BAM_FILE + ".bai")`. Although `BAM_FILE` is otherwise unused here so you could just define `BAI_FILE` instead.